modules/user-access/pages/query-privilege-migration.adoc

```diff
@@ -0,0 +1,136 @@
+
+If users were granted built-in roles with `READ_QUERY` privileges before upgrade, the `READ_QUERY` privileges should be granted to the users directly with all existing queries in the same scope.
+
+====
+.For example:
+* User `Alice` is granted with the built-in role `observer` in graph `recommend` in 3.10.0, and there are two queries `query1` and `query2` in graph `recommend`. After the upgrade process for 4.1.0, user `Alice` should have the `READ_QUERY` privileges on queries `query1` and `query2` in graph `recommend`.
+====
+
+==== Migrate RBAC READ_QUERY privilege for User-Defined roles
+If users were granted user-defined roles with `READ_QUERY` privileges before upgrade, the `READ_QUERY` privileges should be granted to the same role with all existing queries in the same scope.
+
+====
+.For example:
+* User `Bob` is granted with the user-defined role `myRole` with `READ_QUERY` in graph `recommend` in 3.10.0, and there are two queries `query1` and `query2` in graph `recommend`. After the upgrade process for 4.1.0, role `myRole` should have the `READ_QUERY` privileges on query `query1` and `query2` in graph `recommend`. User `Bob` can still read queries `query1` and `query2` since he is still granted with role `myRole`.
+====
+
+[#_migrate_rbac_write_query]
+=== Migrate RBAC WRITE_QUERY privilege
+In our previous versions, `WRITE_QUERY` privilege means we make all changes for a query. Hence, `WRITE_QUERY` privilege should be migrated to all privileges except `OWNERSHIP` for all existing queries, including `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` on each query object and `CREATE_QUERY` on the same scope for `WRITE_QUERY` privilege.
+
+Similarly, this migration process should be different for our existing `WRITE_QUERY` privileges when your users are granted with built-in roles or user-defined roles.
+
+==== Migrate RBAC WRITE_QUERY privilege for Built-in roles
+The `WRITE_QUERY` privilege for built-in roles should not be kept since 4.1.0. For the detailed changes of built-in roles, please refer to the table above.
+
+If users were granted built-in roles with `WRITE_QUERY` privileges before upgrade, following privileges, `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY`, should be granted to the users directly with all existing queries in the same scope.
+
+====
+.For example:
+* User `Alice` is granted with the built-in role `designer` in graph `recommend` in 3.10.0, and there are two queries `query1` and `query2` in graph `recommend`. After the upgrade process for 4.1.0, user `Alice` should hold the `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` privileges on queries `query1` and `query2` in graph `recommend`. At the same time, user `Alice` can still create queries in graph `recommend` since she still holds the role `designer` in graph `recommend`, which provide the `CREATE_QUERY` privilege in graph `recommend`.
```

Comment:
```suggestion
==== Migrate RBAC WRITE_QUERY privilege for Built-in roles
The `WRITE_QUERY` privilege is deprecated in TigerGraph 4.1. For the detailed changes of built-in roles, please refer to the table above.

If users were granted built-in roles with `WRITE_QUERY` privileges before upgrade, the following privileges, `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY`, will be granted to the users directly with all existing queries in the same scope.

====
.For example:
* User `Alice` was granted the built-in role `designer` in graph `recommend` in TigerGraph 3.10, and there are two queries `query1` and `query2` in graph `recommend`. After the upgrade to TigerGraph 4.1, user `Alice` will hold the `READ_QUERY`, `UPDATE_QUERY`, `DROP_QUERY`, `INSTALL_QUERY`, `EXECUTE_QUERY` privileges on queries `query1` and `query2` in graph `recommend`. At the same time, user `Alice` can still create queries in graph `recommend` since she still holds the role `designer` in graph `recommend`, which provides the `CREATE_QUERY` privilege in graph `recommend`.
```